### PR TITLE
Prevent partial reads/writes

### PR DESCRIPTION
**Pull request by paragonie-scott**
Reference: #125

### CODE REVIEW

File: src/File.php [hunk 1]
```diff
@@ -799,4 +773,85 @@ private static function getFileVersionConfigFromMajorMinor($major, $minor)
...
+        $buf = '';
+        $remaining = $num;
+        do {
+            if ($remaining <= 0) {
+                break;
+            }
+            $read = \fread($stream, $remaining);
+            if ($read === false) {
+                throw new Ex\CannotPerformOperationException(
+                    'Could not write to the file'
```
Comment (paragonie-scott) [inline src/File.php]: Oops. Need to fix the text here.

**Comment by paragonie-scott** [comment]
Issues that should be fixed now:

* #114 
* #132 
* #133 
* Possibly #134 (thought we might want to keep that open until audited)

**Comment by defuse** [comment]
Fuck, I want to reply to one of @paragonie-scott's comments, but I can't remember which commit it's in, and GitHub doesn't give me the link.

> No, it doesn't return false. You threw a return; in encryptFile() and the unit tests didn't catch it.

So I accidentally added a return statement somewhere in the function, and that broke it, without breaking the unit tests, and returning true ensures that they'll catch it in the future? Could you document that (or just point me to a commit), I'd like to improve the unit tests so that they'll catch it.

**Comment by defuse** [comment]
@paragonie-scott: I think you addressed all my comments, is this ready to merge?

**Comment by paragonie-scott** [comment]
I believe it's ready.

**Pull request merged**